### PR TITLE
wfnews-1940 wfnews-1952 wfnews-1943 wfnews-1428

### PR DESCRIPTION
**Pull request by yzlucas**

### CODE REVIEW

File: client/wfnews-war/src/main/angular/src/app/components/saved/saved-location-full-details/saved-location-full-details.component.ts [hunk 1]
```diff
@@ -195,8 +195,12 @@ this.agolService
           .then((bans) => {
             if (bans && bans.features) {
               this.fireBans = [];
-              for (const item of bans.features) {
-                this.fireBans.push(item);
+              for (const item of bans?.features) {
```
Comment (dhlevi) [inline client/wfnews-war/src/main/angular/src/app/components/saved/saved-location-full-details/saved-location-full-details.component.ts:198]: Remove the `?` from this line, update the `if` clause above.

```
if (bans?.features) {
   for (const item of bans.features) {
...
```

File: client/wfnews-war/src/main/angular/src/app/components/saved/saved.component.ts [hunk 2]
```diff
@@ -86,9 +86,18 @@ export class SavedComponent implements OnInit {
         .subscribe((bans) => {
           this.savedLocations[outerIndex].bans = [];
           for (const innerIndex in bans?.features) {
-            const element = bans?.features[innerIndex];
-            this.savedLocations[outerIndex].bans.push(element);
-            this.cdr.markForCheck();
+            if (bans?.features.hasOwnProperty(innerIndex)) {
+              const element = bans?.features[innerIndex];
```
Comment (dhlevi) [inline client/wfnews-war/src/main/angular/src/app/components/saved/saved.component.ts:90]: You can remove the `?` here as the if check above already ensures the value exists

`hasOwnProperty` should not be executed on the object directly, but through the Object prototype

`Object.hasOwn(bans?.features, innerIndex)`

File: client/wfnews-war/src/main/angular/src/app/components/saved/saved.component.ts [hunk 3]
```diff
@@ -86,9 +86,18 @@ export class SavedComponent implements OnInit {
...
+          
+              // Check if the attribute already exists in the savedLocations array
+              const isAttributeAlreadyExists = this.savedLocations[outerIndex].bans.some(existingElement =>
```
Comment (dhlevi) [inline client/wfnews-war/src/main/angular/src/app/components/saved/saved.component.ts:93]: maybe `doesAttributeExist` is a better var name? Current one is bit of an weird mouthful. Could also "not" the attribute here so the `if` statement can be truthy

**Review by dhlevi — CHANGES_REQUESTED** [review]

**Comment by yzlucas** [comment]
Moved attributePresent checker into common-utility.service

**Review by ssylver93 — APPROVED** [review]

**Pull request merged**